pynicotine/search.py

```diff
@@ -451,14 +452,69 @@ def create_search_result_list(self, searchterm, wordindex, excluded_words, parti
             log.add_debug("Error: DB closed during search, perhaps due to rescanning shares or closing the application")
             return None
 
-    def process_search_request(self, searchterm, username, token, direct=False):
+    def create_file_info_list(self, results, max_results, permission_level):
+        """ Given a list of file indices, retrieve the file information for each index """
+
+        reveal_buddy_shares = config.sections["transfers"]["reveal_buddy_shares"]
+        reveal_trusted_shares = config.sections["transfers"]["reveal_trusted_shares"]
+        is_buddy = (permission_level == "buddy")
+        is_trusted = (permission_level == "trusted")
+
+        fileinfos = []
+        private_fileinfos = []
+        num_fileinfos = 0
+
+        public_files = core.shares.share_dbs.get("files")
+        buddy_files = core.shares.share_dbs.get("buddyfiles")
+        trusted_files = core.shares.share_dbs.get("trustedfiles")
+
+        if public_files is None or buddy_files is None or trusted_files is None:
```

Comment:
Condition gate not required, because if the database is uninitialized yet this already caught by `if word_index is None` in process_search_requrest()